### DISCUSSION TITLE
I have a websocket already, why would I use subscriptions?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [urql-graphql/urql#2249] [Q&A] [answered]
Hello everyone, using Elixir/ Phoenix so I have access to channels and it's really easy to have real-time communication over sockets, how would I benefit from subscriptions as opposed to just using WebSockets?

**Comment by JoviDeCroock** [comment, marked as answer]
Similar to other GraphQL operations, subscriptions give insight into the shape of your data which makes it possible to automatically update data in a normalized cache. For example, you have a todo-entity in your cache, someone else updates this entity, a subscription would tell our cache to update this and the UI will gracefully react to it. With a manual websocket you'll have to do this yourself.
